modules/oe_translation_poetry/src/Form/PoetryCheckoutFormBase.php

```diff
@@ -0,0 +1,410 @@
+    ];
+
+    $form['actions']['cancel'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Cancel and delete job'),
+      '#button_type' => 'secondary',
+      '#submit' => ['::cancelRequest'],
+      '#limit_validation_errors' => [],
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    // The submit handler is submitRequest().
+  }
+
+  /**
+   * Submits the request to Poetry.
+   *
+   * @param array $form
+   *   The form.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state.
+   */
+  public function submitRequest(array &$form, FormStateInterface $form_state): void {
+    $translator_settings = $this->poetry->getTranslatorSettings();
+    $jobs = $this->queue->getAllJobs();
+    $entity = $this->queue->getEntity();
+    $identifier = $this->poetry->getIdentifierForContent($entity);
+    $identifier->setProduct($this->requestType);
+
+    $date = new \DateTime($form_state->getValue('details')['date']);
+    $formatted_date = $date->format('d/m/Y');
+
+    /** @var \EC\Poetry\Messages\Requests\CreateTranslationRequest $message */
+    $message = $this->poetry->get('request.create_translation_request');
+    $message->setIdentifier($identifier);
+
+    // Build the details.
+    $details = $message->withDetails();
+    $details->setDelay($formatted_date);
+
+    if ($form_state->getValue('details')['comment']) {
+      $details->setRemark($form_state->getValue('details')['comment']);
+    }
+
+    // We use the formatted identifier as the user reference.
+    $details->setClientId($identifier->getFormattedIdentifier());
+    $title = $this->createRequestTitle();
+    $details->setTitle($title);
+    $details->setApplicationId($translator_settings['application_reference']);
+    $details->setReferenceFilesRemark($entity->toUrl()->setAbsolute()->toString());
+    $details
+      ->setProcedure('NEANT')
+      ->setDestination('PUBLIC')
+      ->setType('INTER');
+
+    $message->setDetails($details);
+
+    // Build the contact information.
+    foreach (PoetryTranslatorUI::getContactFieldNames('contact') as $name => $label) {
+      $message->withContact()
+        ->setType($name)
+        ->setNickname($form_state->getValue('details')['contact'][$name]);
+    }
+
+    // Build the organisation information.
+    foreach (PoetryTranslatorUI::getContactFieldNames('organisation') as $name => $label) {
+      $message->withContact()
+        ->setType($name)
+        ->setNickname($form_state->getValue('details')['organisation'][$name]);
+    }
+
+    // Build the return endpoint information.
+    // @todo update once we implemented the notification handling.
+    $return = $message->withReturnAddress();
+    $return->setUser('test');
+    $return->setPassword('test');
+    // The notification endpoint WSDL.
+    $return->setAddress(Url::fromRoute('<front>')->setAbsolute()->toString());
```

Comment:
This, user and password could already be set using config values.

Reply:
Will handle this entire thing in the notifications ticket.